network/p2p/keyTranslator.go

```diff
@@ -92,13 +99,63 @@ func PublicKey(fpk fcrypto.PublicKey) (lcrypto.PubKey, error) {
 		}
 	} else if keyType == lcrypto_pb.KeyType_Secp256k1 {
 		bytes = make([]byte, crypto.PubKeyLenECDSASecp256k1+1) // libp2p requires an extra byte
-		bytes[0] = 4                                           // magic number in libp2p to refer to an uncompressed key
+		bytes[0] = 4                                           // signals uncompressed form as specified in section 4.3.6/7 of ANSI X9.62.
 		copy(bytes[1:], tempBytes)
 	}
 
 	return um(bytes)
 }
 
+// This converts some libp2p PubKeys to a flow PublicKey
+// - the supported key types are ECDSA P-256 and ECDSA Secp256k1 public keys,
+// - libp2p also supports RSA and Ed25519 keys, which Flow doesn't, their conversion will return an error.
+func FlowPublicKeyFromLibP2P(lpk lcrypto.PubKey) (fcrypto.PublicKey, error) {
+
+	switch ktype := lpk.Type(); ktype {
+	case lcrypto_pb.KeyType_ECDSA:
+		pubB, err := lpk.Raw()
+		if err != nil {
+			return nil, lcrypto.ErrBadKeyType
```

Comment:
Maybe making more distinct error types helps to spot the issue upon happening, i.e., this error type is the same as the next case.

Reply:
This error and the next will not actually happen: I'm marshalling the key material and then reinterpreting it as bytes using the underlying x509 library manually, because libp2p deprecated its [`Bytes`](https://github.com/libp2p/go-libp2p-core/pull/204) function. This is just bypassing lack of access to private members.